nixos/modules/services/networking/wireguard.nix

```diff
@@ -160,6 +160,14 @@ let
         interval of 25 seconds; however, most users will not need this.'';
       };
 
+      table = mkOption {
+        default = "main";
+        type = types.str;
+        description = ''The kernel routing table to add this peer's associated
```

Comment:
After pondering this problem for a while, I have to agree. Policy routing via `ip rule` can never achieve its full potential if, in a single interface, different routes to the same CIDR range are disallowed anyway. So supporting per-peer routing tables is not that much of a win. Furthermore, having to specify `table` for each peer (if it's not `main`) makes the "correct" setup needlessly complicated.

Linux even gives you BCP{38,84} filtering for free, with `net.ipv4.conf.{default,all}.rp_filter`! Using a single interface for multiple peers now seems only marginally easier in the "server-like" case. It does save private keys and UDP ports, though.

In any case: I've modified the branch accordingly. eaab02b has the goods.